### ISSUE TITLE
Crash on server startup

### ISSUE THREAD

**Issue by DeadSix27** [MinecraftModDevelopmentMods/BaseMetals#214]
Guess 1.12.2 is not supported ? 

Server crashes on startup with BaseMetals-1.12-2.5.0-beta2.31.jar and OreSpawn-1.12-3.1.0.8.jar and TConstruct.

Not sure if this is a TConstruct issue.
But it looks like your mods loading client classes on the server?

`java.lang.Error: java.lang.reflect.InvocationTargetException`
Stacktrace and modlist: https://gist.github.com/DeadSix27/4d99e19705f54f3163ff9553e2c931ee

____ 
I didn't want to open another issue just for this but:
Nether metals with and without orespawn causes this crash: 
 https://gist.github.com/DeadSix27/f816fb11e175708856977116d59636d1

**Comment by dshadowwolf** [comment]
Looks like something tried to access client code when running on a server, trace indicates something in Tinkers' (possibly) - Tinkers' code in the current release, while mostly functional client-side, has some issues and likely does not work properly server-side. This was not learned until after the release, sadly.

The NetherMetals one puzzles me, as it was tested quite a bit and only needed some API updates to build for 1.12... I'll look into that as soon as time permits. (working on a proper fix for the Tinkers' interactions issue)

**Comment by dshadowwolf** [comment]
New version of Base Metals (and new versions of Base Minerals, End Metals, Modern Metals and Nether Metals) on Curse. The new versions all work together - the releases were because of some major bug-fixes and changes to the core code of the Tinkers' Construct interaction if Base Metals.

Can you give them a try and let us know if they solved the problem ?

**Comment by DeadSix27** [comment]
@dshadowwolf I will later, its early morning here :) Thanks for your work!

**Comment by DeadSix27** [comment]
Ok here's my testing:

Nether/End Metals without Base Metals crashes with this error: https://gist.github.com/DeadSix27/03460050c9781ac41081d941f545c3b7

Also is a dependency name set wrong? Without MMDOrespawn it says it requires "null", log: https://gist.github.com/DeadSix27/ef0ae6522aea8653aa909ba1750d5664

Base Minerals/Metals &or Modern Metals conflict with "moartinkers" @ TConstruct, log: https://gist.github.com/DeadSix27/1948e23001ad3ee8241c18cd272b3357

EDIT: Without moartinkers it starts fine, will report back whether ores spawn fine or anything weird happens later.

**Comment by dshadowwolf** [comment]
At this time BaseMetals is a requirement for both NetherMetals and EndMetals. There is a support library embedded in BaseMetals that is supposed to be separated out at some point.

Unless specifically turned off in BaseMetals.cfg (B:using_orespawn) MMD OreSpawn is also a requirement - none of the *Metals mods contain code for generating ore spawns on their own.

The last is interesting but not unexpected. Moar Tinkers also provides Platinum - this conflicts. I do suppose, however, that we should have some error checking in that area. Will have to look into a proper fix.

**Comment by DeadSix27** [comment]
@dshadowwolf I now run into more issues, specifically TinkersConstruct compability. https://gist.github.com/DeadSix27/3aea1f4eabe3b57596a9ca838f7f8a70

Happens with BaseMinerals and ModernMetals installed.

**Comment by dshadowwolf** [comment]
Just pushed a fix - seems that Tinkers was calling a function for getting a localized string server-side and the function was using code not in the server jar. Fixed it by using the deprecated I18N stuff that is going to disappear at some point.

Won't be in a Curse release for a couple weeks, as there is some backend work in BaseMetals needed before -beta3 gets a release, but the [maven](https://maven.mcmoddev.com) will have a build with the fix.

**Comment by DeadSix27** [comment]
@dshadowwolf Yep, maven one works fine, (with Orespawn,base, nether and end metals)
Btw, is this texture problem a known issue? (also isn't emerald a mineral.. huh do I need BaseMinerals?):
![image](https://user-images.githubusercontent.com/1492505/31574768-7abfcb5c-b0d6-11e7-9f8d-807547607a09.png)

**Comment by dshadowwolf** [comment]
The missing textures? Yeah - its a known issue and some of it is because there are some things being generated that shouldn't be - or have never gotten a texture. There have been some issues caused by some major changes in the internals that attempted to simplify and make things more generic.

**Comment by DeadSix27** [comment]
@dshadowwolf Alright, but ye beside that I have not yet encountered any issues since updating to the latest maven builds

**Comment by dshadowwolf** [comment]
Alright - MMD OreSpawn has been updated quite a bit in the past week with more coming. Several new "features" added - a "vein" generator, a "cloud" ('normal-cloud') and a "cluster" generator - note that the latter two probably don't match what CoFHWorld calls the same thing.

The "cloud" is a set of points distributed using a random input to a normal-distribution function, each point is an ore. The "cluster" uses the same function, but for a smaller cloud and small nodes at each point.

When I've got some time I'll try to document all of that, but right now there is a major change coming to the config files of OreSpawn that'll make them a lot more feature-complete. In the mean-time, this one is fixed, so closing the issue.
